File: src/test/java/me/pcasaes/hexoids/core/domain/utils/TrigUtilTest.java
package me.pcasaes.hexoids.core.domain.utils;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TrigUtilTest {

    @Test
    void testZeroAngleMagnitude() {
        assertEquals(0f, TrigUtil.calculateXComponentFromAngleAndMagnitude(0, 0), Float.MAX_VALUE);
        assertEquals(0f, TrigUtil.calculateYComponentFromAngleAndMagnitude(0, 0), Float.MAX_VALUE);
    }
}

File: src/test/kotlin/me/pcasaes/hexoids/core/domain/utils/TrigUtilTest.kt
package me.pcasaes.hexoids.core.domain.utils

import me.pcasaes.hexoids.core.domain.utils.TrigUtil.xComponentFromAngleAndMagnitude
import me.pcasaes.hexoids.core.domain.utils.TrigUtil.yComponentFromAngleAndMagnitude
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TrigUtilTest {
    
    @Test
    fun testZeroAngleMagnitude() {
        Assertions.assertEquals(0F, xComponentFromAngleAndMagnitude(0F, 0F), Float.Companion.MAX_VALUE)
        Assertions.assertEquals(0F, yComponentFromAngleAndMagnitude(0F, 0F), Float.Companion.MAX_VALUE)
    }
}